tensorboard/program.py

```diff
@@ -441,23 +467,6 @@ def _make_server(self):
         return self.server_class(app, self.flags)
 
 
-def _should_use_data_server(load_fast_flag, logdir):
```

Comment:
FWIW, it seems like we still want this logic if we want to avoid behavior with a data server <0.5.0 regressing to the way it was before #4786, i.e. where `auto` will try to use the data server even if the logdir is not compatible.  I suppose if we see this PR as just supplanting #4786 entirely (such that if you want graceful fallback you need to be on 0.5.0+ rather than having the logic replicated here) that's fine, but just pointing it out.

Reply:
Fair enough. I was intending to supplant, but there’s not much harm in
keeping it for a few more days. Done.